internal/multierror/multierror_test.go

```diff
@@ -0,0 +1,17 @@
+package multierror
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestErrorJoin(t *testing.T) {
+	err := Join(errors.New("foobar"), fmt.Errorf("invalid config:\n%w", Join(errors.New("foo"), errors.New("bar"))))
```

Comment:
If we use this instead:

```
err := Join(errors.New("foobar"), Join(errors.New("foo"), errors.New("bar")))
```

The output will be clumsy:

```
* foobar
* * foo
	* bar
```

This is not great, but I believe it's better to defer to `Join` callers to do the right thing instead (eg. add an error count, a nice preamble message, etc...).

Reply:
The newline is a bit awkward indeed. Just gave it a quick try to see if we can make that automatic, but... it's tricky.